### PR TITLE
Fixed Pedestal not dropping item when is broken or explodes

### PR DESCRIPTION
**Pull request by Redemption198**
Now the pedestal drops the item which is currently above of it when it's broken.
Edit: Changed obstructed message + minor changes.

### CODE REVIEW

**Comment by Poslovitch** [comment]
Don't agree with splitting the obstructed message. This is a useless information to know what is above the pedestal, I think.

**Comment by Redemption198** [comment]
@Poslovitch what about this: "&4Pedestal is obstructed! &cRemove anything above the pedestal!"

**Comment by Poslovitch** [comment]
I agree :)

**Comment by Redemption198** [comment]
@Poslovitch I found another "bug" when a TNT explodes the item which is above the pedestal drops on the floor.



I can try to fix it, but I don't know if the item should be dropped or deleted.

**Comment by Poslovitch** [comment]
add a config option for it

**Comment by TheBusyBiscuit** [comment]
Don't register additional events for Slimefun Blocks, use the BlockBreakHandler instead when calling .register()

It also takes care of explosions.
I also wouldn't add an option for it, just drop it everytime.

File: src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java [hunk 1]
```diff
@@ -171,5 +171,32 @@ public void onBlockPlace(BlockPlaceEvent e) {
...
+			if(item.getID().equalsIgnoreCase("ANCIENT_PEDESTAL")) {
+				Item stack = findItem(b);
+				if(stack == null) return;
+				if(SlimefunStartup.getCfg().getBoolean("options.drop-item-when-ancient-pedestal-explodes"))
+					b.getWorld().dropItem(b.getLocation(), fixItemStack(stack.getItemStack(), stack.getCustomName()));
+				stack.remove();
+			}
+		}
+	}
+}
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java]: for all of the above:

**https://github.com/TheBusyBiscuit/Slimefun4/blob/6f61176d0c9cb944976e95813160a8df0759effc/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/handlers/BlockBreakHandler.java**

Comment (TheBusyBiscuit) [reply src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java]: Nevermind, it's this class:

https://github.com/TheBusyBiscuit/Slimefun4/blob/6f61176d0c9cb944976e95813160a8df0759effc/src/me/mrCookieSlime/Slimefun/Objects/SlimefunBlockHandler.java

Comment (TheBusyBiscuit) [reply src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java]: Could also be this class:

https://github.com/TheBusyBiscuit/Slimefun4/blob/6f61176d0c9cb944976e95813160a8df0759effc/src/me/mrCookieSlime/Slimefun/Objects/SlimefunBlockHandler.java

**Comment by Redemption198** [comment]
@TheBusyBiscuit I don't know if it is what you meant.

**Comment by TheBusyBiscuit** [comment]
Yes, this is exactly where the fix needed to happen.
Thanks, one minor correction:

When comparing non-primitives, use .equals() over ==

**Pull request merged**